### PR TITLE
Feature avatar icon

### LINKED ISSUE #19359: [rfc] icon of avatar support React.ReactNode
- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### What problem does this feature solve?
```
    <Avatar size="small" src={o.avatar} icon={<IconFont type="xx"></IconFont>} />
```

### What does the proposed API look like?
```
    <Avatar size="small" src={o.avatar} icon={<IconFont type="xx"></IconFont>} />
```

### PR DESCRIPTION
**Pull request by DiamondYuan**
### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 💡 Background and solution
close #19359

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  icon of Avatar support ReactNode         |
| 🇨🇳 Chinese |   Avatar 的 icon 属性支持传递   ReactNode      |

### ☑️ Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

-----
[View rendered components/avatar/demo/type.md](https://github.com/ant-design/ant-design/blob/feature-avatar-icon/components/avatar/demo/type.md)
[View rendered components/avatar/index.en-US.md](https://github.com/ant-design/ant-design/blob/feature-avatar-icon/components/avatar/index.en-US.md)
[View rendered components/avatar/index.zh-CN.md](https://github.com/ant-design/ant-design/blob/feature-avatar-icon/components/avatar/index.zh-CN.md)

### CODE REVIEW

**Comment by DiamondYuan** [comment]
close https://github.com/ant-design/ant-design/issues/19359

**Comment by DiamondYuan** [comment]
@afc163  test failed.

image

File: components/avatar/index.zh-CN.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ title: Avatar
 
 | 参数 | 说明 | 类型 | 默认值 | 版本 |
 | --- | --- | --- | --- | --- |
-| icon | 设置头像的图标类型，参考 `Icon` 组件 | string | - |  |
+| icon | 设置头像的图标类型，参考 `Icon` 组件 ,或者是 ReactNode | string \| ReactNode | - |  |
```
Comment (afc163) [inline components/avatar/index.zh-CN.md]:
```suggestion
| icon | 设置头像的图标类型，参考 `Icon` 组件，或者是 ReactNode | string \| ReactNode | - |  |
```

Comment (afc163) [inline components/avatar/index.zh-CN.md]: | icon | 设置头像的图标类型，可设为 Icon 的 `type` 或 ReactNode | string \| ReactNode | - |  |

**Review by afc163 — APPROVED** [review]

**Pull request merged**